File: go/bubble-sort/bubble-sort.go
package bubble_sort

import "cmp"

func Bubble_Sort[T cmp.Ordered](array []T) {
	length := len(array)
	for i, _ := range array {
		for j := 0; j < length-1-i; j++ {
			next_index := j + 1
			if array[j] > array[next_index] {
				current_value := array[j]

				array[j] = array[next_index]
				array[next_index] = current_value
			}
		}
	}
}

File: typescript/src/bubble-sort/bubble-sort.ts
// vim: noexpandtab

/**
 * Mutates and returns sorted array of numbers
 */
export function bubbleSort(array: number[]) {
	for (const i of array.keys()) {
		for (let j = 0; j < array.length - 1 - i; ++j) {
			const next_index = j + 1;
			if (array[j] > array[next_index]) {
				const tmp = array[j];
				array[j] = array[next_index];
				array[next_index] = tmp;
			}
		}
	}
	return array;
}
